lib/puppet/functions/camelcase.rb

```diff
@@ -0,0 +1,62 @@
+# Creates a Camel Case version of a String
+#
+# This function is compatible with the stdlib function with the same name.
+#
+# The function does the following:
+# * For a `String` the conversion replaces all combinations of *_<char>* with an upcased version of the
+#   character following the _.  This is done using Ruby system locale which handles some, but not all
+#   special international up-casing rules (for example German double-s ß is upcased to "Ss").
+# * For an `Iterable[Variant[String, Numeric]]` (for example an `Array`) each value is capitalized and the conversion is not recursive.
+# * If the value is `Numeric` it is simply returned (this is for backwards compatibility).
+# * An error is raised for all other data types.
+# * The result will not contain any underscore characters.
+#
+# Please note: This function relies directly on Ruby's String implementation and as such may not be entirely UTF8 compatible.
+# To ensure best compatibility please use this function with Ruby 2.4.0 or greater - https://bugs.ruby-lang.org/issues/10085.
+#
+# @example Camelcase a String
+# ```puppet
+# 'hello_friend'.camelcase()
+# camelcase('hello_friend')
```

Comment:
Would it be possible to add a second optional boolean to indicate that the first letter should not be upper case? I.e. "hello_friend" becomes "helloFriend" instead of "HelloFriend"? Such conversions are not uncommon.

Reply:
Sure, that would work in a backwards compatible way. May want to add such a feature via some kind of format argument to not block for other formatting options. For example how should acronyms and abbreviations in all caps be handled say 'http_connection' or 'HTTP_connection' is that HTTPConnection or HTTP_Connection. Current implementation will change the case of input like "lowHIGH_Highlow" to "LowhighHighlow". 

This is beyond the scope of this ticket though - write a ticket?